conda/cli/install.py

```diff
@@ -172,120 +172,144 @@ def get_revision(arg, json=False):
-    if isupdate and not (
-        args.file
-        or args.packages
-        or context.update_modifier == UpdateModifier.UPDATE_ALL
-    ):
-        raise CondaValueError(
-            """no package names supplied
-# Example: conda update -n myenv scipy
-"""
-        )
 
+    # Ensure the prefix exists if it is meant to
     if newenv:
-        check_prefix(prefix, json=context.json)
-        if context.subdir != context._native_subdir():
-            # We will only allow a different subdir if it's specified by global
-            # configuration, environment variable or command line argument. IOW,
-            # prevent a non-base env configured for a non-native subdir from leaking
-            # its subdir to a newer env.
-            context_sources = context.collect_all()
-            if context_sources.get("cmd_line", {}).get("subdir") == context.subdir:
-                pass  # this is ok
-            elif context_sources.get("envvars", {}).get("subdir") == context.subdir:
-                pass  # this is ok too
-            # config does not come from envvars or cmd_line, it must be a file
-            # that's ok as long as it's a base env or a global file
-            elif not paths_equal(context.active_prefix, context.root_prefix):
-                # this is only ok as long as it's base environment
-                active_env_config = next(
-                    (
-                        config
-                        for path, config in context_sources.items()
-                        if paths_equal(context.active_prefix, path.parent)
-                    ),
-                    None,
-                )
-                if active_env_config.get("subdir") == context.subdir:
-                    # In practice this never happens; the subdir info is not even
-                    # loaded from the active env for conda create :shrug:
-                    msg = dals(
-                        f"""
-                        Active environment configuration ({context.active_prefix}) is
-                        implicitly requesting a non-native platform ({context.subdir}).
-                        Please deactivate first or explicitly request the platform via
-                        the --platform=[value] command line flag.
-                        """
-                    )
-                    raise OperationNotAllowed(msg)
-            log.info(
-                "Creating new environment for a non-native platform %s",
-                context.subdir,
-            )
+        # new environments should not have a prefix that exists
+        pass
     elif isdir(prefix):
-        delete_trash(prefix)
-        if not isfile(join(prefix, "conda-meta", "history")):
+        # if the prefix exists (and this is not a new environment)
+        # the conda-meta/history file must also exists - if it does not
+        # then this is not a valid conda environment
+        if isfile(join(prefix, "conda-meta", "history")):
```

Comment:
https://github.com/conda/conda/pull/14668, aren't you checking if this file is writable too? Could do both here.

Reply:
Yep! I think #14668 is scoped much more tightly and can be reviewed/merged before this. Will opt to pull those changes in when they get merged :fireworks: